frontend/packages/core-api-client/src/lib/client/Recipient.ts

```diff
@@ -0,0 +1,31 @@
+import { BaseRESTClient } from './BaseRESTClient';
+import {Response} from "../types";
+import {
+  Recipient,
+  RecipientDefinition, RecipientGetRequest,
+  RecipientList, RecipientListRequest
+} from "../types/client/Recipient";
+
+export class RecordClient {
+  restClient: BaseRESTClient;
+
+  constructor(restClient: BaseRESTClient) {
+    this.restClient = restClient;
+  }
+
+  create = (recipient: RecipientDefinition): Promise<Response<RecipientDefinition, Recipient>> => {
+    return this.restClient.post( '/records', recipient);
+  };
```

Comment:
Up for debate, but I prefer the `Response` to be internal to the api and for us to throw an error if it fails. Something like:

```
 create = async (recipient: RecipientDefinition): Promise<Recipient> => {
    const response = await this.restClient.post<RecipientDefinition, Recipient>(
      '/records',
      recipient,
    );
    if (response.success)
      return (response as SuccessResponse<RecipientDefinition, Recipient>).response;
    throw new Error();
  };
```

Reply:
Although actually, this method should just forward the request to the record client.

```
constructor(recordClient: RecordClient) {
    this.recordClient = recordClient;
}

return this.recordClient.create(recipient);
```

Reply:
agreed about keeping responses api-internal. wouldn't throw an error the way things are right now, because the details from the ErrorResponse are needed for validation.
could probably be done by throwing an error too, but as this is blocking multiple tickets and us moving to the native app, I don't think that should be the focus.

Reply:
agreed about keeping responses api-internal. wouldn't throw an error the way things are right now, because the details from the ErrorResponse are needed for validation.
could probably be done by throwing an error too, but as this is blocking multiple tickets and us moving to the native app, I don't think that should be the focus.

Reply:
Yeah happy to keep it as this for now and refactor later. I would typically include the error messages in the error object.